### PR TITLE
Fix test_wait_set.cpp to not wait on uninitialized arrays.

### PR DESCRIPTION
**Pull request by hidmic**
CI up to `test_rmw_implementation` and `rcl`:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12539/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7508/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10251/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12460/)

### CODE REVIEW

File: test_rmw_implementation/test/test_wait_set.cpp [hunk 1]
```diff
@@ -190,94 +237,157 @@ TEST_F(CLASSNAME(TestWaitSet, RMW_IMPLEMENTATION), rmw_wait)
...
+    clients.clients[0] = client->data;
+    events.events[0] = &event;
     ret = rmw_wait(
-      subscriptions, guard_conditions, services, clients, events, wait_set,
+      &subscriptions, &guard_conditions, &services, &clients, &events, wait_set,
       &timeout_argument);
 
-    EXPECT_TRUE(RMW_RET_TIMEOUT == ret || RMW_RET_ERROR == ret);
-    rmw_reset_error();
+    if (RMW_RET_TIMEOUT == ret) {
```
Comment (Lobotuerk) [inline test_rmw_implementation/test/test_wait_set.cpp:381]: If `rmw_wait` times out, is it granted that all 5 parameters are going to be already received aka nullptrs?

**Review by Lobotuerk — APPROVED** [review]

**Review by ahcorde — APPROVED** [review]

File: test_rmw_implementation/test/test_wait_set.cpp (hunk 1 above)
Comment (hidmic) [reply test_rmw_implementation/test/test_wait_set.cpp:381]: A `nullptr` means they have not been received. I expect none to be activated.

**Comment by hidmic** [comment]
I'll merge into #139. We can discuss further there.

File: test_rmw_implementation/test/test_wait_set.cpp (hunk 1 above)
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_wait_set.cpp:381]: Oh, i mised those, then its fine

**Pull request merged**